src/libLookup/Lookup.cpp

```diff
@@ -3372,8 +3372,13 @@ void Lookup::FindMissingMBsForLastNTxBlks(const uint32_t& num) {
   }
 }
 
-const vector<Transaction>& Lookup::GetTxnFromShardMap(uint32_t index) {
-  return m_txnShardMap[index];
+const vector<Transaction>& Lookup::GetTxnFromShardMap(uint32_t shardId) const {
```

Comment:
This could be overkill. No need to find it always before fetching. Because empty value `vector` is assigned if key `index` does not exist & we any ways check for empty at caller. Note this is always called at https://github.com/Zilliqa/Zilliqa/blob/6157d930d4098b0c58f0fe02846f4e7e432957c4/src/libServer/LookupServer.cpp#L260

Also, if we stick with additional checks, we need to have it also at https://github.com/Zilliqa/Zilliqa/blob/6157d930d4098b0c58f0fe02846f4e7e432957c4/src/libLookup/Lookup.cpp#L5330 otherwise entry will be created before its cleared.

Reply:
It is ok to return empty vector, but the change of this function is to make get function to be const.